Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,580 @@
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _(
+          'Displays a rich text label using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
+        ),
+        'JsPlatform/Extensions/bbcode32.png',
+        objectBBText
+      )
+      .setIncludeFile('Extensions/BBText/bbtextruntimeobject.js')
+      .addIncludeFile('Extensions/BBText/bbtextruntimeobject-pixi-renderer.js')
+      .addIncludeFile('Extensions/BBText/dist/pixi-multistyle-text.umd.js');
+
+    // Utility function to add both a setter and a getter to a property from a list. Useful for setting multiple generic properties
+    const addSettersAndGettersToObjectHelper = (
+      gdObject,
+      properties,
+      objectName
+    ) => {
+      properties.forEach(property => {
+        const parameterType =
+          property.type === 'boolean' ? 'yesorno' : property.type;
+
+        // Add all the generic GETTERS
+        if (parameterType === 'number') {
+          gdObject
+            .addExpression(
+              `Get${property.functionName}`,
+              _('Get ') + property.description,
+              _('Get ') + property.description,
+              _(''),
+              '',
+              `${property.iconPath}.png`,
+              `${property.iconPath}.png`
+            )
+            .addParameter(
+              'object',
+              _(`${objectName} object`),
+              objectName,
+              false
+            )
+            .getCodeExtraInformation()
+            .setFunctionName(`get${property.functionName}`);
+        } else {
+          gdObject
+            .addStrExpression(
+              `Get${property.functionName}`,
+              _('Get ') + property.description,
+              _('Get ') + property.description,
+              _(''),
+              '',
+              `${property.iconPath}.png`,
+              `${property.iconPath}.png`
+            )
+            .addParameter(
+              'object',
+              _(`${objectName} object`),
+              objectName,
+              false
+            )
+            .getCodeExtraInformation()
+            .setFunctionName(`get${property.functionName}`);
+        }
+
+        // Add all the generic SETTERS
+        if (parameterType === 'number' || parameterType === 'string') {
+          gdObject
+            .addAction(
+              `Set${property.functionName}`,
+              _('Set ') + property.description,
+              _('Set ') + property.description,
+              _('Set ') +
```

Comment:
I would rather have a sentence that is in the same format as the rest of the software. (`Do _PARAM1__PARAM2_ to the xxx of yyy`).

I understand that there must be a better/more english way to write these, but let's do this in another card: https://trello.com/c/0CfDCJIk/248-make-events-sheet-actions-conditions-more-english-like-set-x-to-y-add-y-to-x-y-is-equal-to-x :) 
Having consistency in all the software will allow to do a global search/replace if we find something better. The solution might be also to have another layer "translating" sentences to more english like sentences, but again let's do this in another PR/update.